src/inputs/ChipSelectField.tsx

```diff
@@ -319,3 +323,29 @@ type ListBoxSection<O> = { title: string; options: O[]; isPersistent?: boolean }
 export function isListBoxSection<O>(obj: O | ListBoxSection<O>): obj is ListBoxSection<O> {
   return typeof obj === "object" && "options" in obj;
 }
+
+interface CreateNewFieldProps {
+  onBlur: Callback;
+  onEnter: (value: string) => {};
+}
+
+// Wrapper for the ChipTextField used in the "Create New" flow on ChipSelectField
+function CreateNewField(props: CreateNewFieldProps) {
+  const { onBlur, onEnter } = props;
+  const [value, setValue] = useState<string>("Add new");
+  const tid = useTestIds(props);
+
+  return (
+    <ChipTextField
+      autoFocus
+      label="Add new"
+      value={value}
+      onChange={setValue}
+      onEnter={() => onEnter(value)}
+      onBlur={onBlur}
+      {...tid}
+    />
+  );
+}
+
+let firstRender = true;
```

Comment:
question: Is this down at the global scope? How does that work when there are more than one chip select rendered?

Reply:
Great question and thank you for flagging! I had assumed this would be in the module scope so that it'd be isolated per component. But you are right, this seems to bleed further out. I will update to be a ref internal to the component.